src/lib/parsers/bundles.ts

```diff
@@ -0,0 +1,167 @@
+        let itemID: string = bundleRequirementsDataSplit[i];
+        let itemQuantity: number = parseInt(bundleRequirementsDataSplit[i + 1]);
+        let itemQuality = bundleRequirementsDataSplit[i + 2] as ItemQuality;
+
+        // An item ID of -1 is used for gold for the vault room
+        if (itemID == "-1") {
+          let currentBundleItem: BundleItem = {
+            itemID: "-1",
+            itemQuantity: itemQuantity,
+            itemQuality: itemQuality,
+          };
+          requiredItems.push(currentBundleItem);
+          continue;
+        }
+
+        let currentBundleItem: BundleItem = {
+          itemID: itemID,
+          itemQuantity: itemQuantity,
+          itemQuality: itemQuality,
+        };
+
+        requiredItems.push(currentBundleItem);
+      }
+
+      let itemData = null;
+      if (rewardDataSplit[0] == "O") {
+        itemData = typedObjects[rewardDataSplit[1]];
+      }
+      if (rewardDataSplit[0] == "BO") {
+        itemData = typedBigCraftables[rewardDataSplit[1]];
+      }
+
+      // Ideally we'd pull reward data for other types from their respective data files,
+      // but until those data files exist we'll just hardcode existing objects outside
+      // of objects and big_craftables.
+      if (!itemData) {
+        // The missing bundle has an empty string for rewardDataString, so we need some
+        // custom reward logic.
+        if (rewardDataString == "") {
+          itemData = {
+            name: "Movie Theater unlock",
+          };
+        }
+        if (rewardDataSplit[0] == "R" && rewardDataSplit[1] == "517") {
+          itemData = {
+            name: "Small Glow Ring",
+          };
+        }
+        if (rewardDataSplit[0] == "R" && rewardDataSplit[1] == "518") {
+          itemData = {
+            name: "Small Magnet Ring",
+          };
+        }
+        // Apparently the heater isn't defined in big_craftables, presumably because it's
+        // not craftable.
+        if (rewardDataSplit[0] == "BO" && rewardDataSplit[1] == "104") {
+          itemData = {
+            name: "Heater",
+          };
+        }
+      }
+
+      let currentBundle: Bundle = {
+        name: bundleName,
+        areaName: areaName,
+        localizedName: localizedBundleName ? localizedBundleName : bundleName,
+        color: parseInt(colorId),
+        items: requiredItems,
+        itemsRequired: requiredItemCount,
+        bundleReward: {
+          itemType: rewardDataSplit[0] || "",
+          itemID: rewardDataSplit[1] || "-1",
+          itemQuantity: parseInt(rewardDataSplit[2] || "1"),
+        },
+      };
+
+      let returnableBundle: BundleWithStatus = {
+        bundle: currentBundle,
+        bundleStatus: bundleCompletionData[bundleId] || [],
+      };
+
+      bundles.push(returnableBundle);
+    }
+
+    return bundles;
+  } catch (e) {
+    if (e instanceof Error)
+      throw new Error(`Error in parseBundles: ${e.message}`);
```

Comment:
Add logging here

Reply:
+1, add on what bundle / item / etc. more helpful to the player & our sentry logging